File: first/firstsync/src/main/java/joe/sync/mapper/FirstMapper.java
package joe.sync.mapper;

import joe.model.FirstDto;
import joe.reactive.second.dto.SecondDto;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface FirstMapper {
    default SecondDto map(FirstDto firstDto) {
        return SecondDto.builder()
                .field11(firstDto.getField1())
                .field22(firstDto.getField2())
                .field33(firstDto.getField3())
                .field44(firstDto.getField4())
                .needCallback(firstDto.isNeedCallback())
                .build();
    }
}

File: first/firstredis/src/main/kotlin/redis/mapper/FirstMapper.kt
package redis.mapper

import joe.model.FirstDto
import joe.reactive.second.dto.SecondDto
import org.springframework.stereotype.Component

@Component
class FirstMapper {
    fun map(firstDto: FirstDto): SecondDto {
        return SecondDto.builder()
            .field11(firstDto.field1)
            .field22(firstDto.field2)
            .field33(firstDto.field3)
            .field44(firstDto.field4)
            .build()
    }
}